component/backup.jsonnet

```diff
@@ -0,0 +1,126 @@
+  stringData: {
+    username: params.backup.bucket.accesskey,
+    password: params.backup.bucket.secretkey,
+  },
+};
+local bucketSecretRef = {
+  name: bucketSecret.metadata.name,
+  accesskeyname: 'username',
+  secretkeyname: 'password',
+};
+
+local schedule = backup.Schedule(
+  params.name,
+  params.backup.schedule,
+  keep_jobs=params.backup.keepjobs,
+  bucket=params.backup.bucket.name,
+  backupkey=backupSecretRef,
+  s3secret=bucketSecretRef,
+  create_bucket=false,
+).schedule + backup.PruneSpec('23 * * * *', 30, 20) {
+  metadata+: {
+    namespace: params.namespace,
+  },
+};
+
+local backupConfig = kube.ConfigMap('%s-backup' % params.name) {
+  metadata+: {
+    namespace: params.namespace,
+  },
+  data: {
+    'vault-agent-config.hcl': ''
+                              + 'exit_after_auth = false'
+                              + '\n      auto_auth {'
+                              + '\n          method "kubernetes" {'
+                              + '\n              config = {'
+                              + '\n                  role = "backup"'
+                              + '\n              }'
+                              + '\n          }'
+                              + '\n          sink "file" {'
+                              + '\n              config = {'
+                              + '\n                  path = "/home/vault/.vault-token"'
+                              + '\n                  mode = 0644\n              }'
+                              + '\n          }'
+                              + '\n      }',
+  },
+};
+
+local backupSA = kube.ServiceAccount('%s-backup' % params.name) {
+  metadata+: {
+    namespace: params.namespace,
+  },
+};
+
+local backupPod = backup.PreBackupPod(
+  params.name,
+  '%s/%s:%s' % [ params.images.vault.registry, params.images.vault.repository, params.images.vault.version ],
+  'vault operator raft snapshot save /dev/stdout',
+  fileext='.snapshot'
+) {
+  metadata+: {
+    namespace: params.namespace,
+  },
+  spec+: {
+    pod+: {
+      spec+: {
+        containers: [
+          super.containers[0] {
+            env: [
+              {
+                name: 'HOME',
+                value: '/home/vault',
+              },
+              {
+                name: 'VAULT_ADDR',
+                value: 'http://%s-active:8200' % params.name,
+              },
+              {
+                name: 'SKIP_SETCAP',
+                value: 'true',
+              },
+            ],
+            volumeMounts: [
+              {
+                name: 'config',
+                mountPath: '/etc/vault/',
+              },
+            ],
+          },
+        ],
+        serviceAccountName: backupSA.metadata.name,
+        volumes: [
+          {
+            name: 'config',
+            configMap: {
+              name: backupConfig.metadata.name,
+            },
+          },
+        ],
+      },
+    },
```

Comment:
You may be able to leverage kube-libsonnet's `PodSpec()` helper here, cf. https://github.com/bitnami-labs/kube-libsonnet/blob/20bf70d8d24b3a50baaae8f951dcbc62a66b464f/kube.libsonnet#L291-L334

With that you'd get the ability to use the underscore-postfixed variants of `containers` and `env`, e.g. for env, you could then do:

```
env_: {
  HOME: '/home/vault',
  VAULT_ADDR: 'http://%s-active:8200' % params.name,
  SKIP_SETCAP: 'true',
}
```

instead of having to write the plain K8s env array.